recipes/data_bag.rb

```diff
@@ -25,23 +25,29 @@
   raise "Could not find data bag for #{node['ucarp']['data_bag']['cluster']}"
 end
 
-if platform_family?('rhel')
-  template "/etc/ucarp/vip-#{ucarp_databag['vip_id']}.conf" do
-    source 'vip.conf.erb'
-    owner 'root'
-    group 'root'
-    mode '0644'
-    variables ucarp_databag.to_hash
-    notifies :restart, 'service[ucarp]'
-  end
+file "/etc/ucarp/vip-#{ucarp_databag['vip_id']}.pwd" do
+  content ucarp_databag['password']
+  owner 'root'
+  group 'root'
+  mode '0400'
+  notifies :restart, 'service[ucarp]'
+end
+
+template "/etc/ucarp/vip-#{ucarp_databag['vip_id']}.conf" do
+  source 'vip.conf.erb'
+  owner 'root'
+  group 'root'
+  mode '0644'
+  variables ucarp_databag.to_hash
+  notifies :restart, 'service[ucarp]'
+end
 
-  service 'ucarp' do
-    case node['ucarp']['init_type']
-    when 'systemd'
-      provider Chef::Provider::Service::Systemd
-      service_name "ucarp@vip-#{ucarp_databag['vip_id']}.service"
-    end
-    supports status: true, restart: true
-    action [:start, :enable]
+service 'ucarp' do
+  case node['ucarp']['init_type']
+  when 'systemd'
+    provider Chef::Provider::Service::Systemd
+    service_name "ucarp@vip-#{ucarp_databag['vip_id']}.service"
   end
+  supports status: true, restart: true
+  action [:start, :enable]
```

Comment:
service cannot be started without being enabled first

Reply:
I don't think that's correct. Enabling the service just means that it comes on automatically on startup.

Reply:
@TravisWhitehead is correct